cmd/minikube/cmd/start_flags.go

```diff
@@ -443,6 +444,12 @@ func getRepository(cmd *cobra.Command, k8sVersion string) string {
 		download.SetAliyunMirror()
 	}
 
+	// Need to override the default of old kubeadm versions
+	v := semver.MustParse(strings.TrimPrefix(k8sVersion, version.VersionPrefix))
+	if repository == "" && images.NeedsImageRepository(v) {
+		repository = images.DefaultKubernetesRepo(v)
```

Comment:
This breaks kicbase, it sets the image repository, which also sets the image repository for the kicbase. So using an old version of Kubernetes that hits this line results in trying to download `registry.k8s.io/k8s-minikube/kicbase-builds:v0.0.37-1677262057-15923`.

Reply:
I don't think this is need either, the Kubernetes images uses the repo from `DefaultKubernetesRepo()` which since your change is always going to be `NewDefaultKubernetesRepo`.

I removed this statement and tried it with k8s v1.16.0 and it looks like it's using the new registry
```
I0328 15:08:25.263619   61896 image.go:134] retrieving image: registry.k8s.io/kube-proxy:v1.16.0
I0328 15:08:25.263673   61896 image.go:134] retrieving image: registry.k8s.io/coredns:1.6.2
I0328 15:08:25.263852   61896 image.go:134] retrieving image: registry.k8s.io/pause:3.1
I0328 15:08:25.263930   61896 image.go:134] retrieving image: gcr.io/k8s-minikube/storage-provisioner:v5
I0328 15:08:25.264030   61896 image.go:134] retrieving image: registry.k8s.io/kube-apiserver:v1.16.0
I0328 15:08:25.264790   61896 image.go:134] retrieving image: registry.k8s.io/kube-scheduler:v1.16.0
I0328 15:08:25.264896   61896 image.go:134] retrieving image: registry.k8s.io/kube-controller-manager:v1.16.0
I0328 15:08:25.264997   61896 image.go:134] retrieving image: registry.k8s.io/etcd:3.3.15-0
I0328 15:08:25.271738   61896 image.go:177] daemon lookup for registry.k8s.io/coredns:1.6.2: Error: No such image: registry.k8s.io/coredns:1.6.2
I0328 15:08:25.271978   61896 image.go:177] daemon lookup for registry.k8s.io/kube-proxy:v1.16.0: Error: No such image: registry.k8s.io/kube-proxy:v1.16.0
I0328 15:08:25.273085   61896 image.go:177] daemon lookup for registry.k8s.io/kube-apiserver:v1.16.0: Error: No such image: registry.k8s.io/kube-apiserver:v1.16.0
I0328 15:08:25.273154   61896 image.go:177] daemon lookup for gcr.io/k8s-minikube/storage-provisioner:v5: Error: No such image: gcr.io/k8s-minikube/storage-provisioner:v5
I0328 15:08:25.274976   61896 image.go:177] daemon lookup for registry.k8s.io/pause:3.1: Error: No such image: registry.k8s.io/pause:3.1
I0328 15:08:25.275102   61896 image.go:177] daemon lookup for registry.k8s.io/kube-controller-manager:v1.16.0: Error: No such image: registry.k8s.io/kube-controller-manager:v1.16.0
I0328 15:08:25.275555   61896 image.go:177] daemon lookup for registry.k8s.io/etcd:3.3.15-0: Error: No such image: registry.k8s.io/etcd:3.3.15-0
I0328 15:08:25.275967   61896 image.go:177] daemon lookup for registry.k8s.io/kube-scheduler:v1.16.0: Error: No such image: registry.k8s.io/kube-scheduler:v1.16.0
```